### DISCUSSION TITLE
自定义 OAuth 后，如何在 `@waline/vercel` 中自定义 `ALLOW_SOCIALS`

### DISCUSSION THREAD

**Discussion by maxchang3** [walinejs/waline#1870] [Q&A] [answered]
您好，最近从上古版本的 waline 更新到了最新版。我自己部署了 OAuth。

我可以在 OAuth 的 index.html 中

```js
window.ALLOW_SOCIALS = ['github']
```

来屏蔽其他登录选项。

但是因为使用了`@waline/vercel`，登录界面仍然是由 `@waline/vercel` 中的 `@waline/admin` 提供的，我该如何修改 `ALLOW_SOCIALS` 呢？🤔是否有提供相应接口？多次查阅文档未发现。

另外，OAuth 也引入了 `@waline/admin `。这样就有两个登录前台了，我应该分别如何对待他们呢？

感谢！

**Comment by lizheming** [comment, marked as answer]
1. 你可以将 `/ui` 路由到你自己的静态页上实现 `window.ALLOW_SOCIALS` 注入，目前还木有打通环境变量到前端的注入所以这块暂时还没有便捷的注入方式，感兴趣的话可以提个 PR
2. OAuth 的那个忽略即可，那个是我用来申请 QQ 社交登录加的。QQ 社交登录申请需要有完整的网站，而且必须先把 QQ 登录功能做好才行，于是我就只能… 不过还是失败了，我还得再试试申请一下

**Reply by maxchang3** [reply]
了解！感谢回复！
